### PR TITLE
[USING] Add Joinmarket Explanation

### PR DESCRIPTION
**Pull request by MaxHillebrand**
[Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver/) is a beautiful mixing protocol, a pioneering service with ingenious economic incentives. It should definitely be explain in the docs, not just because it is good to learn about it, but also because there is also a long term goal to get it implemented in Wasabi. 

This is a ready for review branch to create a new chapter `/using-wasabi/Joinmarket.md` which has a general explanation of the system. FAQs may be added in a separate PR, referencing this main body of work.

### CODE REVIEW

File: docs/using-wasabi/Joinmarket.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+---
+
+# Joinmarket
+
+[[toc]]
+
+## Introduction and History
+
+[Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver) is a CoinJoin protocol implemented in a wallet software which requires Bitcoin Core as backend.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
[Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver) is a CoinJoin protocol implemented in a wallet software which requires Bitcoin Core as the back end.
```
Webster's dictionary & Wikipedia consistently separate backend into back end, and I think correct grammar is to place the article:the in front of back end. Apparently hyphenating (back-end) is an acceptable option also.

File: docs/using-wasabi/Joinmarket.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Its novel innovation is to introduced incentive structures to encourage users to mix their coins.
+One market taker can pay many market makers to create a tailor made CoinJoin.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
One market taker can pay many market makers to create a tailor-made CoinJoin.
```

File: docs/using-wasabi/Joinmarket.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Joinmarket is well [documentation](https://github.com/JoinMarket-Org/joinmarket-clientserver/tree/master/docs) in regards for [installation](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/INSTALL.md), usage of the [command line interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/USAGE.md) and [graphical user interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/JOINMARKET-QT-GUIDE.md).
+
+## Market Makers
+
+Joinmarket makers are users running the [yield generator script](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/YIELDGENERATOR.md) and offering their coins as potential inputs in a CoinJoin transaction.
+These users advertise the total volume of bitcoin they would like to mix [for example a range from 0.1 to 5 bitcoin], as well as the fee they charge for their provided service [for example at least 2000 sats, or 0.02% of the CoinJoin denomination] to a public IRC channel.
+The requirement is an always on, stable and relatively high bandwidth internet connection, so to successfully coordinate many mixes.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The requirement is an always on, stable and relatively high bandwidth internet connection, so as to successfully coordinate many mixes.
```

File: docs/using-wasabi/Joinmarket.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+# Joinmarket
+
+[[toc]]
+
+## Introduction and History
+
+[Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver) is a CoinJoin protocol implemented in a wallet software which requires Bitcoin Core as backend.
+Its novel innovation is to introduced incentive structures to encourage users to mix their coins.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Its novel innovation is to introduce incentive structures to encourage users to mix their coins.
```

File: docs/using-wasabi/Joinmarket.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+These users advertise the total volume of bitcoin they would like to mix [for example a range from 0.1 to 5 bitcoin], as well as the fee they charge for their provided service [for example at least 2000 sats, or 0.02% of the CoinJoin denomination] to a public IRC channel.
+The requirement is an always on, stable and relatively high bandwidth internet connection, so to successfully coordinate many mixes.
+
+:::tip Reclaim your privacy & stack sats!
+Joinmarket makers get paid to do tailor-made CoinJoins with the taker.
+:::
+
+## Market Takers
+
+Joinmarket takers are users who coordinate a tailor made CoinJoin, and they pay the makers to provide the liquidity.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Joinmarket takers are users who coordinate a tailor-made CoinJoin, and they pay the makers to provide the liquidity.
```

File: docs/using-wasabi/Joinmarket.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+Takers analyze the range and fees of offered coins of makers, specify a transaction where the makers coins are part of the input, and the output creates equal value coins.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Takers analyze the range and fees of coins offered by makers, specify a transaction where the makers coins are part of the input, and the output creates equal value coins.
```
This reads a bit better, IMHO. Chaining "ofs" can be hard to follow. (At least for me :) )

File: docs/using-wasabi/Joinmarket.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+The number of makers and the value of the outputs is completely up to the taker, it is a tailor made CoinJoin.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The number of makers and the value of the outputs is completely up to the taker, it is a tailor-made CoinJoin.
```

File: docs/using-wasabi/Joinmarket.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+Although makers can not anonymity the input to output link of other makers, the taker knows exactly which maker gets which output.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Although makers can not connect the input to output link of other makers, the taker knows exactly which maker gets which output.
```
Not sure if "connect" is optimal, but anonymity doesn't work here.

File: docs/using-wasabi/Joinmarket.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+:::tip Create CoinJoins just how you like them!
+Joinmarket takers pay for coordinating tailor-made mixes.
+:::
+
+## Joinmarket + Wasabi
+
+The advantage of Joinmarket is that the taker can coordinate tailor-made mixes, and that the maker gets paid to join.
+The problem with Joinmarket is that the taker can spy on the makers.
+The advantage of Wasabi is that the coordinator cannot spy, and that mixes achieve high anonset.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The advantage of Wasabi is that the coordinator cannot spy, and that Wasabi mixes achieve high anonset. 
```

File: docs/using-wasabi/Joinmarket.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+## Joinmarket + Wasabi
+
+The advantage of Joinmarket is that the taker can coordinate tailor-made mixes, and that the maker gets paid to join.
+The problem with Joinmarket is that the taker can spy on the makers.
+The advantage of Wasabi is that the coordinator cannot spy, and that mixes achieve high anonset. 
+The problem with Wasabi is that there are standardized equal value output amounts.
+
+:::tip A perfect match!
+The problems with Wasabi are the strenght of Joinmarket, and the downsides of Joinmarket are the features of Wasabi!
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The problems with Wasabi are the strength of Joinmarket, and the downsides of Joinmarket are the features of Wasabi!
```

File: docs/using-wasabi/Joinmarket.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+:::
+
+These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi wallet, for example such a protocol:
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi wallet. For example, such a protocol:
```

File: docs/using-wasabi/Joinmarket.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+1. With your non-private coins, make at least two rounds of Wasabi zero link CoinJoins.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
1. With your non-private coins, make at least two rounds of Wasabi ZeroLink CoinJoins.
```

File: docs/using-wasabi/Joinmarket.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+This means that your pre-mix coins are not tied to the following Joinmarket mixers.
+
+2. With these anonset coins, run the yield generator as a Joinmarket maker for as long as possible, partaking in many mixes with many different takers.
+In this step you gain privacy that you were part of the initial Wasabi mix.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
In this step you gain privacy, in that you were part of the initial Wasabi mix.
```

File: docs/using-wasabi/Joinmarket.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+Further, the mixing fees you earn might be enough to pay at least for the transaction and coordination fees of the whole process.
+
+3. With these Joinmarket coins, make at least two rounds of Wasabi zero link CoinJoins.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
3. With these Joinmarket coins, make at least two rounds of Wasabi ZeroLink CoinJoins.
```

File: docs/using-wasabi/Joinmarket.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+This will break the Joinmarket history of these coins, thus nobody knows you were a maker.
+
+4. With these high anonset coins, coordinate a tailor made CoinJoin as a Joinmarket taker, where you can specify exactly the value you want to actually send.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
4. With these high anonset coins, coordinate a tailor-made CoinJoin as a Joinmarket taker, where you can specify exactly the value you want to actually send.
```

File: docs/using-wasabi/Joinmarket.md (hunk 11 above)
Comment (MaxHillebrand) [reply docs/using-wasabi/Joinmarket.md]:
```suggestion
These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi wallet.
For example such a protocol:
```

I added the new line.

File: docs/using-wasabi/Joinmarket.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+[[toc]]
+
+## Introduction and History
+
+[Joinmarket](https://github.com/JoinMarket-Org/joinmarket-clientserver) is a CoinJoin protocol implemented in a wallet software which requires Bitcoin Core as the back end.
+Its novel innovation is to introduce incentive structures to encourage users to mix their coins.
+One market taker can pay many market makers to create a tailor-made CoinJoin.
+The software consists of several python scripts, has a graphical user interface, and utilizes IRC communication channels.
+The initial v0.1.0 release was in December 2015 by [Chris Belcher](https://github.com/chris-belcher), and since then it has gathered steady support by several contributors, most notably [Adam Gibson](https://github.com/AdamISZ).
+Joinmarket is an integral part of the Bitcoin privacy infrastructure, and the research and implementation by the contributors is invaluable.
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Joinmarket is an integral part of the Bitcoin privacy infrastructure, the research and implementation by the contributors are invaluable.
```

File: docs/using-wasabi/Joinmarket.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+Joinmarket is well [documentation](https://github.com/JoinMarket-Org/joinmarket-clientserver/tree/master/docs) in regards for [installation](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/INSTALL.md), usage of the [command line interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/USAGE.md) and [graphical user interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/JOINMARKET-QT-GUIDE.md).
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Joinmarket is well [documented](https://github.com/JoinMarket-Org/joinmarket-clientserver/tree/master/docs) in regards to [installation](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/INSTALL.md), usage of the [command line interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/USAGE.md) and [graphical user interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/JOINMARKET-QT-GUIDE.md).
```

File: docs/using-wasabi/Joinmarket.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+:::tip Reclaim your privacy & stack sats!
+Joinmarket makers get paid to do tailor-made CoinJoins with the taker.
+:::
+
+## Market Takers
+
+Joinmarket takers are users who coordinate a tailor-made CoinJoin, and they pay the makers to provide the liquidity.
+Takers analyze the range and fees of coins offered by makers, specify a transaction where the makers coins are part of the input, and the output creates equal value coins.
+The number of makers and the value of the outputs is completely up to the taker, it is a tailor-made CoinJoin.
+Although makers can not connect the input to output link of other makers, the taker knows exactly which maker gets which output.
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Although makers cannot link the input to the output of other makers, the taker knows exactly which maker gets which output.
```

File: docs/using-wasabi/Joinmarket.md [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+
+:::tip Create CoinJoins just how you like them!
+Joinmarket takers pay for coordinating tailor-made mixes.
+:::
+
+## Joinmarket + Wasabi
+
+The advantage of Joinmarket is that the taker can coordinate tailor-made mixes, and that the maker gets paid to join.
+The problem with Joinmarket is that the taker can spy on the makers.
+The advantage of Wasabi is that the coordinator cannot spy, and that Wasabi mixes achieve high anonset.
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The advantage of Wasabi is that the coordinator cannot spy, and that Wasabi mixes achieve high anonset.
```

File: docs/using-wasabi/Joinmarket.md [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+
+## Joinmarket + Wasabi
+
+The advantage of Joinmarket is that the taker can coordinate tailor-made mixes, and that the maker gets paid to join.
+The problem with Joinmarket is that the taker can spy on the makers.
+The advantage of Wasabi is that the coordinator cannot spy, and that Wasabi mixes achieve high anonset. 
+The problem with Wasabi is that there are standardized equal value output amounts.
+
+:::tip A perfect match!
+The problems with Wasabi are the strength of Joinmarket, and the downsides of Joinmarket are the features of Wasabi!
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
The problems with Wasabi are the strengths of Joinmarket, and the downsides of Joinmarket are the features of Wasabi!
```

File: docs/using-wasabi/Joinmarket.md [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+:::
+
+These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi wallet.
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi Wallet.
```

File: docs/using-wasabi/Joinmarket.md [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+Joinmarket is an integral part of the Bitcoin privacy infrastructure, and the research and implementation by the contributors is invaluable.
+
+Joinmarket is well [documentation](https://github.com/JoinMarket-Org/joinmarket-clientserver/tree/master/docs) in regards for [installation](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/INSTALL.md), usage of the [command line interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/USAGE.md) and [graphical user interface](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/JOINMARKET-QT-GUIDE.md).
+
+## Market Makers
+
+Joinmarket makers are users running the [yield generator script](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/YIELDGENERATOR.md) and offering their coins as potential inputs in a CoinJoin transaction.
+These users advertise the total volume of bitcoin they would like to mix [for example a range from 0.1 to 5 bitcoin], as well as the fee they charge for their provided service [for example at least 2000 sats, or 0.02% of the CoinJoin denomination] to a public IRC channel.
```
Comment (yahiheb) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
These users advertise the total volume of bitcoin they would like to mix [for example a range from 0.1 to 5 bitcoins], as well as the fee they charge for their provided service [for example at least 2000 sats, or 0.02% of the CoinJoin denomination] to a public IRC channel.
```

File: docs/using-wasabi/Joinmarket.md (hunk 16 above)
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
Joinmarket is an integral part of the Bitcoin privacy infrastructure. 
The research and implementation by the contributors are invaluable.
```
Fixes a "run-on" sentence.

Comment (yahiheb) [reply docs/using-wasabi/Joinmarket.md]: Superseded by https://github.com/zkSNACKs/WasabiDoc/pull/474#discussion_r365557771

**Comment by MaxHillebrand** [comment]
Thanks @davterra & @yahiheb for the review!!

I think this is a pretty decent first approach to the chapter, so I think we are getting ready for merging of this PR, with smaller PRs to fine tune and add further content. 

However, before I merge this, may I bring it to the attention of @chris-belcher and @AdamISZ, you are both credited in this chapter for your monumental contributions, if you have the time, please give a quick ACK/NACK to what is written.

**Comment by AdamISZ** [comment]
@MaxHillebrand thanks; for my part, everything looks correct (including and especially the links), except one historical detail: Joinmarket launched on mainnet iirc in May 2015 (and indeed was in active use very quickly); although I understand you've taken the first official release date, which is perfectly logical. Anyway @chris-belcher will correct me if I'm wrong.

> Although makers can not anonymity the input to output link of other makers, the taker knows exactly which maker gets which output.

Typo here "can not anonymity".

File: docs/using-wasabi/Joinmarket.md [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+These two tools can be combined in great effect, thus it is on the long term road-map to implement Joinmarket functionality in Wasabi Wallet.
+For example such a protocol:
+
+1. With your non-private coins, make at least two rounds of Wasabi ZeroLink CoinJoins.
+This means that your pre-mix coins are not tied to the following Joinmarket mixers.
+
+2. With these anonset coins, run the yield generator as a Joinmarket maker for as long as possible, partaking in many mixes with many different takers.
+In this step you gain privacy, in that you were part of the initial Wasabi mix.
+Further, the mixing fees you earn might be enough to pay at least for the transaction and coordination fees of the whole process.
+
```
Comment (chris-belcher) [inline docs/using-wasabi/Joinmarket.md]: This whole idea seems based on a faulty premise. These downsides of joinmarket only apply if the user is running a maker but most users who actually want privacy use taker mode. Taker mode means no other peer can spy on them, hence there's no point using Wasabi at all in this situation if they've already used joinmarket.

And actually, the coordinator in Wasabi *can* spy in a targeted manner on a specific coin if they do a sybil attack.

Comment (MaxHillebrand) [reply docs/using-wasabi/Joinmarket.md]: Thanks Chris, your of course right that the taker is very private, as he's the coordinator.
What I wanted to point out here is that one round of zero link provides higher anonset as one round of a taker. And intuitively I would say that encapsulating the maker mixes with zero link [or many taker mixes] is a good idea, as here the maker history is hidden from pre-mix sender and post-mix receiver. Further, this is layering on defenses, so IF only Joinmarket, or only Wasabi, are broken, then the other is still providing privacy. 

Anyhow, this is only one of potentially many interesting strategies one can make when combining Joinmarket with ZeroLink, maybe this chapter is a nice place to elaborate on these.

The potential sybil attack of the coordinator is another good point, though I think this deserves its own chapter, not in this one on Joinmarket.

Comment (chris-belcher) [reply docs/using-wasabi/Joinmarket.md]: It is worthwhile changing the part about JoinMarket makers and takers, because right now the text implies that the only way to use JoinMarket is to be a maker, which is not correct.

Comment (AdamISZ) [reply docs/using-wasabi/Joinmarket.md]: @chris-belcher I don't really understand what you mean with that last comment. The very first line of the text you're reviewing says "The advantage of Joinmarket is that the taker can coordinate tailor-made mixes ..." so how is it implying that the only way to use Joinmarket is as a maker?

Otoh I do agree on reflection that the line "The problem with Joinmarket is that the taker can spy on the makers." is just kinda wrong .. a maker chooses to offer coins for liquidity knowing that any taker that chooses them knows the linkage .. they're not worse off than before, so "the taker spies on the makers" is the wrong way to look at it - sorta.

This point generally is pretty tricky: people often come into Joinmarket and think maker side is obviously the smart way to use it, and it's not **entirely** wrong to see it that way - for some use cases, it can make sense, but there should be no illusion of full privacy in doing that.

Comment (MaxHillebrand) [reply docs/using-wasabi/Joinmarket.md]: Please see 5acc940, I tried to explain the trade-offs more accurately, also that many rounds of making brings privacy against any one taker.

File: docs/using-wasabi/Joinmarket.md [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+## Market Makers
+
+Joinmarket makers are users running the [yield generator script](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/YIELDGENERATOR.md) and offering their coins as potential inputs in a CoinJoin transaction.
+These users advertise the total volume of bitcoin they would like to mix [for example a range from 0.1 to 5 bitcoins], as well as the fee they charge for their provided service [for example at least 2000 sats, or 0.02% of the CoinJoin denomination] to a public IRC channel.
+A maker chooses to offer coins for liquidity knowing that the taker can deanonymze the maker in this one transaction, thus it is adviced to do several rounds of making with different takers to gain privacy even against any one taker.
```
Comment (davterra) [inline docs/using-wasabi/Joinmarket.md]:
```suggestion
A maker chooses to offer coins for liquidity knowing that the taker can deanonymze the maker in this one transaction, thus it is advised to do several rounds of making with different takers to gain privacy even against any one taker. 
```
